week4/class_version/src/yahtzee_itertools.py

```diff
@@ -0,0 +1,73 @@
+"""
+Planner for Yahtzee
+Simplifications:  only allow discard and roll, only score against upper level
+"""
+
+from itertools import chain, combinations, product
+
+def score(hand):
```

Comment:
This is where OO comes in handy: you don't need to repeat yourself.

Fine for now, though.